infra/Pos/DHT/Real/Real.hs

```diff
@@ -132,27 +147,27 @@ getKnownPeersImpl
        , Bi DHTData
        , Bi DHTKey
        )
-    => KademliaDHT m [DHTNode]
-getKnownPeersImpl = do
-    myId <- currentNodeKey
-    (inst, initialPeers, explicitInitial) <-
-        KademliaDHT $
-        (,,) <$> ask <*> asks kdiInitialPeers <*> asks kdiExplicitInitial
+    => KademliaDHTInstance
+    -> m [DHTNode]
+kademliaGetKnownPeers inst = do
+    let myId = kdiKey inst
+    let initialPeers = kdiInitialPeers inst
+    let explicitInitial = kdiExplicitInitial inst
     peers <- liftIO $ K.dumpPeers $ kdiHandle inst
     let initPeers = bool [] initialPeers explicitInitial
     filter ((/= myId) . dhtNodeId) <$> extendPeers inst myId initPeers peers
   where
-    extendPeers inst myId initial peers =
+    extendPeers inst_ myId initial peers =
```

Comment:
Probably wanted to avoid linter warnings about shadowing. I could just remove the `inst` parameter here.